internal/debug_disabled.go

```diff
@@ -0,0 +1,8 @@
+//go:build !debug
+// +build !debug
+
+package internal
+
+// BuildTypeIsDebug is true if built with -tags debug. This value exists as a constant
+// so the Go compiler can deadcode eliminate reflect from non-debug builds.
+const BuildTypeIsDebug = false
```

Comment:
I think we already have a constant for this: https://github.com/fyne-io/fyne/blob/master/app/app_debug.go#L8C12-L8C12

Other than that, this is a nice improvement!

Reply:
We can't use that field directly as it isn't exported. We can however do `fyne.CurrentApp().Settings().BuildType() == fyne.BuildDebug` but that's quite a slow lookup and I'm not sure to which degree it manages to deadcode eliminate it. I think Andrew cached the debug value like he did to avoid the app lookup + settings + buildtype lookup in the hot rendering path.

Reply:
I wonder if we should move all the build type compile-time constants here then - seems a bit like an anti-pattern to have them defined twice in two different packages. Up to you if you want to address it in this PR now or just merge it and revisit later

Reply:
I agree with you. I'll create a separate package for it, I think. A package like `internal/buildtype` could also have a common way to look up if we are running on Wayland or not for example.

Reply:
I believe this should be fixed now. I'm very happy with the solution personally. Might be worth it to extend to mobile as well but need to see if it manages to deadcode that or not first